### DISCUSSION TITLE
Q: Can the stock be shared?

### DISCUSSION THREAD

**Discussion by ghost** [RGB-WG/rgb#256] [Q&A] [answered]
I'm implementing a Stock based on PostgreSQL as the storage facility, which includes all the implementations from memory.rs. Now,  i have a question:

Currently, each user stores their own stock-related files (state, stash, index). When converting this to database tables, it's natural to have an owner field to distinguish whose stock it is.

I am asking if i can remove the owner field and have all users share a global stock?

**Comment by dr-orlovsky** [comment]
Of course the stock can be shared; however stock properties are fully depend on the specific data providers the stock uses.

First of all, the default `MemStock` and other `Mem*` providers coming with `rgb-std` are not intended to be used on servers. They are designed for a single-user apps, like mobile or desktop wallets, or servers with a single stock for all contracts (like in `rgbex`).

Second, they are not really designed to work with databases or anything which can support partial state saving. Instead, a new dedicated providers has to be developed by companies, according to their needs.

> I am asking if i can remove the owner field and have all users share a global stock?

No, this is against the design I am describing above. You should not touch `Stock` and instead do your own custom providers (like `PsqlStash`, `PsqlState`, `PsqlIndex`) and initialize stock with them like `Stock<PsqlStash, PsqlState, PsqlIndex>`. If you do the providers properly you would get `Stock` object `Sync` or `Send`, as (if) needed.

**Reply by ghost** [reply]
@dr-orlovsky 

I am asking if i can remove the owner field and have all users share a global stock?

My description may not be accurate， i means that if the index, state and stash with no owner key word, so the stock they collected is a global one, dose it have problems?

**Reply by dr-orlovsky** [reply, marked as answer]
> I am asking if i can remove the owner field and have all users share a global stock?

there is no owner keyword in index, state or stash. I do not understand what you refer to.
